src/raftstore/store/snap.rs

```diff
@@ -61,10 +66,120 @@ impl Display for SnapKey {
     }
 }
 
-/// Name prefix for the self-generated snapshot file.
-const SNAP_GEN_PREFIX: &'static str = "gen";
-/// Name prefix for the received snapshot file.
-const SNAP_REV_PREFIX: &'static str = "rev";
+pub enum SendSnapshotFileReader {
```

Comment:
can we use a trait instead of enum, so that we can implement concrete class separately?